server/server_test.go

```diff
@@ -1823,6 +1824,161 @@ allow if {
+			payload:           mustGZIPPayload(util.MustMarshalJSON(generateJSONBenchmarkData(100, 100))),
+			expRespHTTPStatus: 200,
+			maxLen:            1024,
+			gzipMaxLen:        134217728,
+		},
+		{
+			note:              "basic, gzip, large payload, expect reject on gzip payload length",
+			wantGzip:          true,
+			payload:           mustGZIPPayload(util.MustMarshalJSON(generateJSONBenchmarkData(100, 100))),
+			expRespHTTPStatus: 400,
+			maxLen:            1024,
+			gzipMaxLen:        10,
+			expErrorMsg:       "gzip payload too large",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.note, func(t *testing.T) {
+			ctx := context.Background()
+			store := inmem.New()
+			txn := storage.NewTransactionOrDie(ctx, store, storage.WriteParams)
+			examplePolicy := `package example.authz
+
+import rego.v1
+
+default allow := false # Reject requests by default.
+
+allow if {
+	# Logic to authorize request goes here.
+	input.body.user == "alice"
+}
+`
+
+			if err := store.UpsertPolicy(ctx, txn, "test", []byte(examplePolicy)); err != nil {
+				panic(err)
+			}
+
+			if err := store.Commit(ctx, txn); err != nil {
+				panic(err)
+			}
+
+			opts := [](func(*Server)){
+				func(s *Server) {
+					s.WithStore(store)
+				},
+			}
+
+			// Set defaults for max_length configs, if not specified in the test case.
+			if test.maxLen == 0 {
+				test.maxLen = defaultMaxLen
+			}
+			if test.gzipMaxLen == 0 {
+				test.gzipMaxLen = defaultGzipMaxLen
+			}
+
+			f := newFixtureWithConfig(t, fmt.Sprintf(`{"server":{"decision_logs": %t, "decoding":{"max_length": %d, "gzip": {"max_length": %d}}}}`, true, test.maxLen, test.gzipMaxLen), opts...)
+
+			// Forcibly replace the size trailer field for the gzip blob.
+			// Byte order is little-endian, field is a uint32.
+			if test.forcePayloadSizeField != 0 {
+				binary.LittleEndian.PutUint32(test.payload[len(test.payload)-4:], test.forcePayloadSizeField)
+			}
+
+			// execute the request
+			req := newReqV1(http.MethodPost, "/data/test", string(test.payload))
+			if test.wantGzip {
+				req.Header.Set("Content-Encoding", "gzip")
+			}
+			if test.forceContentLen > 0 {
+				req.Header.Set("Content-Length", strconv.FormatInt(test.forceContentLen, 10))
+			}
+			f.reset()
+			f.server.Handler.ServeHTTP(f.recorder, req)
+			if f.recorder.Code != test.expRespHTTPStatus {
+				t.Fatalf("Unexpected HTTP status code, (exp,got): %d, %d", test.expRespHTTPStatus, f.recorder.Code)
+			}
+			if test.expErrorMsg != "" {
+				var serverErr types.ErrorV1
+				if err := json.Unmarshal(f.recorder.Body.Bytes(), &serverErr); err != nil {
+					t.Fatalf("Could not deserialize error message: %s, message was: %s", err.Error(), f.recorder.Body.Bytes())
+				}
+				if !strings.Contains(serverErr.Message, test.expErrorMsg) {
+					t.Fatalf("Expected error message to have message '%s', got message: '%s'", test.expErrorMsg, serverErr.Message)
+				}
+			}
+		})
+	}
+}
+
+// TODO(philipc): Update or delete these tests.
```

Comment:
Do you plan to resolve this as part of this change?

Reply:
I believe the new tests (and tests from the prior PR) *should* be sufficient for testing this change, so I've removed the TODO comments. I had thought at first that I'd need to do renovations on those tests, but it seems they're fine to leave as-is.